fix: i18n keys

KaKi87 opened the pull request:
### What does it do?

It fixes i18n keys in JS that are outdated compared to the English JSON files.

### Why is it needed?

i18n doesn't work when keys don't match a string from the JSON files.

### How to test it?

These changes were automatically generated by the following Bun script that I wrote :

```js
import { $ } from 'bun';
const
  keys = (await Promise.all((await $`find . -type f -name "en.json"`.text()).trim().split('\n').map(i18nPath => (async () => Object.keys((await import(i18nPath)).default))()))).flat(),
  invalidKeys = [];
let replacedKeyCount = 0;
for(const path of (await $`find ./packages -type f -name "*.ts*" -o -name "*.js*"`.text()).trim().split('\n')){
  const content = await Bun.file(path).text();
  let editedTsContent;
  for(const key of new Set([...content.matchAll(new RegExp(/getTranslation[\n ]*?\([\n ]*?['"`]([a-z0-9.-]+?)['"`][\n ]*?\)/gsi))].map(_ => _[1]))){
    if(keys.includes(key)) continue;
    const replacementKey = keys.find(_key => _key.endsWith(key));
    if(replacementKey){
      editedTsContent = (editedTsContent || content).replace(new RegExp(`(?<=['"\`])${key}(?=['"\`])`, 'g'), replacementKey);
      replacedKeyCount++;
    }
    else
      invalidKeys.push({
        path,
        key
      });
  }
  if(editedTsContent)
    await Bun.write(path, editedTsContent);
}
console.log(`
Automatically replaced ${replacedKeyCount} keys.
Remaining invalid keys :
${invalidKeys.map(({ path, key }) => `- ${path}: ${key}`).join('\n')}
`);
```

Which returns the following output as of d100408 :

```
Automatically replaced 126 keys.
Remaining invalid keys :
- ./packages/plugins/i18n/admin/src/components/CMEditViewCopyLocale.tsx: CMEditViewCopyLocale.ModalConfirm.content
- ./packages/core/admin/ee/admin/src/content-manager/pages/ListView/constants.ts: containers.ListPage.table-headers.reviewWorkflows.stage
- ./packages/core/admin/ee/admin/src/content-manager/pages/ListView/constants.ts: containers.ListPage.table-headers.reviewWorkflows.assignee
- ./packages/core/admin/ee/admin/src/content-manager/pages/ListView/constants.ts: containers.ListPage.table-headers.reviewWorkflows.assignee.label
- ./packages/core/admin/admin/src/content-manager/pages/ListView/components/TableCells/Relations.tsx: DynamicTable.relation-loaded
- ./packages/core/admin/admin/src/content-manager/pages/EditSettingsView/components/ModalForm.jsx: containers.SettingPage.editSettings.size.label
- ./packages/core/admin/admin/src/content-manager/components/DynamicZone/DynamicComponent.tsx: components.DynamicZone.more-actions
- ./packages/core/admin/admin/src/content-manager/components/DynamicZone/DynamicComponent.tsx: components.DynamicZone.add-item-above
- ./packages/core/admin/admin/src/content-manager/components/DynamicZone/DynamicComponent.tsx: components.DynamicZone.add-item-below
```

### Related issue(s)/PR(s)

N/A

---

cc. @derrickmehaffy

Thanks

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/strapi?pullRequest=19643) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/strapi?pullRequest=19643) before we can accept your contribution.<br/><hr/>**KaKi87** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/strapi?pullRequest=19643) it.</sub>

derrickmehaffy commented on packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:
```diff
@@ -262,7 +262,7 @@ const CloneDragItem = ({ children, dragHandleTopMargin }: CloneDragItemProps) =>
         role="button"
         aria-label={formatMessage({
-          id: getTranslation('components.DragHandle-label'),
+          id: getTranslation('content-manager.content-manager.components.DragHandle-label'),
```
This one might not be correct

derrickmehaffy commented on packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:
```diff
@@ -228,7 +228,7 @@ const DragAndDropElement = ({
             tabIndex={0}
             aria-label={formatMessage({
-              id: getTranslation('components.DragHandle-label'),
+              id: getTranslation('content-manager.content-manager.components.DragHandle-label'),
```
Same might not be correct

derrickmehaffy commented on packages/core/admin/admin/src/content-manager/components/ContentTypeFormWrapper.tsx:
```diff
@@ -335,7 +335,7 @@ const ContentTypeFormWrapper = ({
         toggleNotification({
           type: 'success',
-          message: { id: getTranslation('success.record.save') },
+          message: { id: getTranslation('content-manager.content-manager.success.record.save') },
```
Same

derrickmehaffy reviewed (changes requested):
Script might have broken some correct values

KaKi87 commented:
Good catch, I'll get right on that.

KaKi87 commented:
The `replaceAll` function was called twice when a key appears twice in the same file. I reverted the initial changes, fixed the script and reran it, which also revealed a few more occurences.

joshuaellis commented on packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:
```diff
@@ -228,7 +228,7 @@ const DragAndDropElement = ({
             tabIndex={0}
             aria-label={formatMessage({
-              id: getTranslation('components.DragHandle-label'),
+              id: getTranslation('content-manager.components.DragHandle-label'),
```
this is incorrect, the `getTranslation` function appends `content-manager` to these strings.

KaKi87 replied:
Oh, where can I detect this ?

derrickmehaffy replied:
I'm guessing that prefix append is broken then as I was looking at this with him in Discord and setting the keys to override doesn't do anything, it always takes the static default value.

KaKi87 replied:
I don't know about `content-manager.components.DragHandle-label` but `content-manager.containers.ListPage.displayedFields` at least was indeed stuck with the default value.

derrickmehaffy commented:
I tested this again with @KaKi87 @joshuaellis and indeed we found where that prefix [should happen](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/content-manager/utils/translations.ts) but regardless translation key overrides are broken:

![image](https://github.com/strapi/strapi/assets/8593673/672325ca-895e-480a-a628-86b8f0ecfa1f)

So this PR might not be valid but the translation key overrides are certainly broken

joshuaellis commented:
> So this PR might not be valid but the translation key overrides are certainly broken

that's not very good is it? 😦

alexandrebodin commented:
Hi, I'm going to close this PR as staled, Let us know if you want to pick up this topic again in the future. THank you for your contribution!

KaKi87 commented:
The topic is still an issue : i18n is currently not working, even though this PR doesn't fix it. Thanks

alexandrebodin commented:
@KaKi87 can you share the corresponding issue link please ? I'm unclear on what is not working exactly

KaKi87 commented:
I didn't meant *issue* as *GitHub issue*, but as *problem*, sorry :sweat_smile: 

As Derrick summarized it well : *translation key overrides are broken*.

alexandrebodin commented:
I understood :D I'm just asking if you have a corresponding github issue so we can track and eventually fix this ;)

KaKi87 commented:
Ah, there isn't one, sorry

alexandrebodin commented:
If you don't mind creating one and add some more info or examples the team will be able to triage it at some point 🙏

KaKi87 commented:
Honestly, since v4 will remain out of Strapi's TODO list until v5 becomes stable, and will then remain low after that, I'm not very motivated to fill the template at this time. 😅

alexandrebodin commented:
No worries it's fine, we have cleaned things in v5 but it's probably not fixing the root issue so it will be opened later during v5
